# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/BrowserChild.cpp
# Commit: beeda5ef3712
# Full Hash: beeda5ef3712b73a19c7a4ce08a52ab4dd08b802
# Author: David P <daparks@mozilla.com>
# Date: 2024-07-04 16:09:52
# Regressor Bug: 1893119
# File Overlap Count: 1
# Description:
#   Bug 1893119: Part 20 - Move EndDragSession from nsIDragService to nsIDragSession r=gstoll,rkraesig,win-reviewers,geckoview-reviewers,m_kato
#   
#   This is a straightforward move.  It does take the liberty of breaking out an
#   EndDragSessionImpl method, which will be needed later.
#   
# ==============================================================================

diff -r 6c3ffc8c534c -r beeda5ef3712 dom/ipc/BrowserChild.cpp
--- a/dom/ipc/BrowserChild.cpp	Thu Jul 04 07:48:10 2024 +0000
+++ b/dom/ipc/BrowserChild.cpp	Thu Jul 04 07:48:11 2024 +0000
@@ -2000,22 +2000,18 @@
     const bool& aDoneDrag, const bool& aUserCancelled,
     const LayoutDeviceIntPoint& aDragEndPoint, const uint32_t& aKeyModifiers,
     const uint32_t& aDropEffect) {
-  nsCOMPtr<nsIDragService> dragService =
-      do_GetService("@mozilla.org/widget/dragservice;1");
-  if (dragService) {
-    RefPtr<nsIDragSession> dragSession = GetDragSession();
-    if (dragSession) {
-      if (aUserCancelled) {
-        dragSession->UserCancelled();
-      }
-
-      RefPtr<DataTransfer> dataTransfer = dragSession->GetDataTransfer();
-      if (dataTransfer) {
-        dataTransfer->SetDropEffectInt(aDropEffect);
-      }
-      dragSession->SetDragEndPoint(aDragEndPoint.x, aDragEndPoint.y);
+  RefPtr<nsIDragSession> dragSession = GetDragSession();
+  if (dragSession) {
+    if (aUserCancelled) {
+      dragSession->UserCancelled();
     }
-    dragService->EndDragSession(aDoneDrag, aKeyModifiers);
+
+    RefPtr<DataTransfer> dataTransfer = dragSession->GetDataTransfer();
+    if (dataTransfer) {
+      dataTransfer->SetDropEffectInt(aDropEffect);
+    }
+    dragSession->SetDragEndPoint(aDragEndPoint.x, aDragEndPoint.y);
+    dragSession->EndDragSession(aDoneDrag, aKeyModifiers);
   }
   return IPC_OK();
 }